aviatrix/resource_aviatrix_spoke_gateway.go

```diff
@@ -1220,6 +1226,18 @@ func resourceAviatrixSpokeGatewayCreate(d *schema.ResourceData, meta interface{}
 		}
 	}
 
+	enableSpokePreserveAsPath := d.Get("enable_preserve_as_path").(bool)
+	if enableSpokePreserveAsPath {
+		if enableBgp {
+			err := client.EnableSpokePreserveAsPath(gateway)
+			if err != nil {
+				return fmt.Errorf("could not enable spoke preserve as path: %v", err)
+			}
+		} else {
+			return fmt.Errorf("enable_preserve_as_path only work for the BGP spoke gateway")
```

Comment:
This could be worded better. Something like :`enable_preserve_as_path is not supported for Non-BGP Spoke Gateways`